pytest_splinter/plugin.py

```
@@ -321,14 +325,19 @@ def make_screenshot_on_failure():
                 else:
                     screenshot_dir = tmpdir.mkdir('screenshots').strpath
                 screenshot_path = os.path.join(screenshot_dir, screenshot_file_name)
-                browser.driver.save_screenshot(screenshot_path)
-                with open(screenshot_path) as fd:
-                    if slaveoutput is not None:
-                        slaveoutput.setdefault('screenshots', []).append({
-                            'class_name': classname,
-                            'file_name': screenshot_file_name,
-                            'content': fd.read()
-                        })
+                try:
+                    LOGGER.info('Saving screenshot to %s' % (screenshot_path,))
+                    browser.driver.save_screenshot(screenshot_path)
+                    with open(screenshot_path) as fd:
+                        if slaveoutput is not None:
+                            slaveoutput.setdefault('screenshots', []).append({
+                                'class_name': classname,
+                                'file_name': screenshot_file_name,
+                                'content': fd.read()
+                            })
+                except URLError as e:
+                    LOGGER.warn('Could not save screenshot: %s' % (e,))
```

yep, it makes sense to catch all Exceptions with a warning
yes, just a note about what happens if screenshot generation fails, and also add what's the most common problem (that the browser is not running anymore)